src/display/pattern_helper.js

```diff
@@ -322,63 +322,58 @@ var TilingPattern = (function TilingPatternClosure() {
 
       var x0 = bbox[0], y0 = bbox[1], x1 = bbox[2], y1 = bbox[3];
 
-      var topLeft = [x0, y0];
-      // we want the canvas to be as large as the step size
-      var botRight = [x0 + xstep, y0 + ystep];
+      // It is a waste of memory to use the size as specified by xstep or ystep
+      // if the bounding box is smaller.
+      xstep = Math.min(xstep, x1);
```

Comment:
I don't think so (correct me if I'm wrong):
The drawing is clipped below (by `clipBbox`) by the square (x0,y0) - (x1,y1). So the canvas itself must be at least as large as (x1,y1).

Using the difference `x1 - x0` would be possible if we can translate the viewport and clip at the rectangle (0,0) - (x1-x0, y1-y0). I haven't tried this yet, do you think that this is a valid operation?